### PR TITLE
feat(store): expose connection pool tuning options

### LINKED ISSUE #628: response store: write performance tuning
## Context

Sub-issue of #3.

Expose connection pool tuning options and optional write buffering for the response store backends (SQLite, PostgreSQL) to handle high request volume.

### Connection pool tuning

Expose configuration options in the store config:
- `max_connections` — maximum number of connections in the pool
- `idle_timeout` — how long idle connections are kept alive

### Bounded write queue (stretch)

If write throughput becomes a bottleneck, add a bounded write queue with:
- `num_writers` — configurable worker count
- `max_write_queue_size` — queue depth limit
- Batched upserts to amortize transaction overhead
- Backpressure policy: configurable drop vs. block behavior
- Graceful drain on shutdown (flush pending writes before exit)

## Acceptance criteria

- [ ] Pool tuning options are exposed in store config and documented
- [ ] Example config updated in `examples/configs/`
- [ ] Unit tests for new config options
- [ ] Integration tests under concurrent write load

### PR DESCRIPTION
**Pull request by leseb**
## Summary

Adds a shared `PoolConfig` struct (`max_connections`, `min_connections`, `idle_timeout_secs`, `acquire_timeout_secs`) to both `ResponseStoreConfig` and `ConversationsConfig`, letting operators tune the sqlx connection pool per backend. All fields are optional — when omitted, sqlx defaults apply. Validation rejects `max_connections: 0` at config load time. SQLite in-memory databases remain pinned to one connection regardless of pool settings.

## Related issue

Closes #628

## Validation

```
make lint   # clippy, nightly fmt, machete, dep/separator/filter-doc/example/markdown lints — all pass
make test   # 12 test suites, 3609 tests, 0 failures
```

New tests cover: `PoolConfig` deserialization (all/partial/empty/unknown fields), duration conversion edge cases (`idle_timeout_secs: 0` disables, `None` uses default), validation (rejects `max_connections: 0`, accepts positive and omitted), and filter-level config round-trips with pool options.

## Checklist

- [x] I reviewed every changed line and can explain the change.
- [x] Tests are added or updated when behavior changes.
- [x] New capabilities include an example config and functional example test.
- [x] User-facing behavior and generated documentation are updated.
- [ ] Performance-sensitive changes include appropriate benchmark or load-test evidence.
- [x] Commits are signed and include a `Signed-off-by` trailer.

## Breaking changes

None. The `pool` field is optional and defaults to `None`, preserving existing behavior.

### CODE REVIEW

File: apis/src/store/pool.rs [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+
+impl PoolConfig {
+    /// Reject invalid values that would cause confusing runtime
+    /// failures (e.g. a zero-connection pool that deadlocks on the
+    /// first query).
+    pub(crate) fn validate(&self) -> Result<(), String> {
+        if self.max_connections == Some(0) {
+            return Err("pool.max_connections must be at least 1".into());
+        }
+        Ok(())
```
[the remark this answers is not in the document]
Comment (leseb) [reply apis/src/store/pool.rs:69]: https://github.com/praxis-proxy/ai/pull/632/commits/63aeb85f42494b1dae17c29722a7002721879b24

File: apis/src/store/pool.rs [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+                None
+            } else {
+                Some(Duration::from_secs(secs))
+            }
+        })
+    }
+
+    /// Convert `acquire_timeout_secs` to a [`Duration`].
+    pub(crate) fn acquire_timeout(&self) -> Option<Duration> {
+        self.acquire_timeout_secs.map(Duration::from_secs)
```
[the remark this answers is not in the document]
Comment (leseb) [reply apis/src/store/pool.rs:87]: https://github.com/praxis-proxy/ai/pull/632/commits/63aeb85f42494b1dae17c29722a7002721879b24

**Review by franciscojavierarceo — APPROVED** [review]
